src/server/scripts/Kalimdor/zone_desolace.cpp

```diff
@@ -233,9 +233,488 @@ class npc_dalinda : public CreatureScript
+    NPC_LESSER_INFERNAL = 4676,
+    NPC_DOOMWARDER = 4677
+};
+
+#define NUM_AMBUSHES_BODYGUARD_FOR_HIRE 3
+#define NUM_KOLKAR_AMBUSHERS 4
+
+#define NUM_AMBUSHES_GIZELTON_CARAVAN 3
+#define NUM_DEMON_AMBUSHERS 3
+
+/// @todo Add values for orientation
+const std::pair<KolkarAmbusher, Position> KOLKAR_SPAWNPOINTS[NUM_AMBUSHES_BODYGUARD_FOR_HIRE][NUM_KOLKAR_AMBUSHERS] = // QUEST_BODYGUARD_FOR_HIRE
+{
+    {
+        std::make_pair(NPC_KOLKAR_WAYLAYER, Position(-969.05f, 1174.91f, 90.39f)),
+        std::make_pair(NPC_KOLKAR_WAYLAYER, Position(-983.01f, 1192.88f, 90.01f)),
+
+        std::make_pair(NPC_KOLKAR_AMBUSHER, Position(-985.71f, 1173.95f, 91.02f)),
+        std::make_pair(NPC_KOLKAR_AMBUSHER, Position(-965.51f, 1193.58f, 92.15f))
+    },
+
+    {
+        std::make_pair(NPC_KOLKAR_WAYLAYER, Position(-1147.83f, 1180.87f, 91.38f)),
+        std::make_pair(NPC_KOLKAR_WAYLAYER, Position(-1160.97f, 1201.36f, 93.15f)),
+
+        std::make_pair(NPC_KOLKAR_AMBUSHER, Position(-1163.96f, 1183.72f, 93.79f)),
+        std::make_pair(NPC_KOLKAR_AMBUSHER, Position(-1146.20f, 1199.75f, 91.37f))
+    },
+
+    {
+        std::make_pair(NPC_KOLKAR_WAYLAYER, Position(-1277.78f, 1218.56f, 109.30f)),
+        std::make_pair(NPC_KOLKAR_WAYLAYER, Position(-1289.25f, 1239.20f, 108.79f)),
+
+        std::make_pair(NPC_KOLKAR_AMBUSHER, Position(-1292.65f, 1221.28f, 109.99f)),
+        std::make_pair(NPC_KOLKAR_AMBUSHER, Position(-1272.91f, 1234.39f, 108.14f))
+    }
+};
+
+const std::pair<DemonAmbusher, Position> DEMON_SPAWNPOINTS[NUM_AMBUSHES_GIZELTON_CARAVAN][NUM_DEMON_AMBUSHERS] = // QUEST_GIZELTON_CARAVAN
+{
+    {
+        std::make_pair(NPC_NETHER_SORCERESS, Position(-1823.7f, 2060.88f, 62.0925f, 0.252946f)),
+        std::make_pair(NPC_LESSER_INFERNAL, Position(-1814.46f, 2060.13f, 62.4916f, 2.00647f)),
+        std::make_pair(NPC_DOOMWARDER, Position(-1814.87f, 2080.6f, 63.6323f, 6.13453f))
+    },
+
+    {
+        std::make_pair(NPC_NETHER_SORCERESS, Position(-1782.92f, 1942.55f, 60.2205f, 3.8722f)),
+        std::make_pair(NPC_LESSER_INFERNAL, Position(-1786.5f, 1926.05f, 59.7502f, 3.2329f)),
+        std::make_pair(NPC_DOOMWARDER, Position(-1805.74f, 1942.77f, 60.791f, 0.829597f))
+    },
+
+    {
+        std::make_pair(NPC_NETHER_SORCERESS, Position(-1677.56f, 1835.67f, 58.9269f, 1.30894f)),
+        std::make_pair(NPC_LESSER_INFERNAL, Position(-1675.66f, 1863.0f, 59.0008f, 0.4827f)),
+        std::make_pair(NPC_DOOMWARDER, Position(-1692.31f, 1862.69f, 58.9553f, 1.45773f))
+    }
+};
+
+class npc_cork_gizelton : public CreatureScript
+{
+public:
+    npc_cork_gizelton() : CreatureScript("npc_cork_gizelton") { }
+
+    struct npc_cork_gizeltonAI : public npc_escortAI
+    {
+        npc_cork_gizeltonAI(Creature* creature) : npc_escortAI(creature)
+        {
+            memset(&summons, 0, sizeof(summons));
+        }
+
+        void InitializeAI() override
+        {
+            _playerGUID = ObjectGuid();
+            headNorth = true;
```

Comment:
this field must be initialized in the constructor too to avoid static analysis issues